tests/test_main.py

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env python
+
+import unittest
+import numpy as np
+from mecode import G
+
+
+g = G(
+    print_lines=False,
+    aerotech_include=False,
+    direct_write=True,
+    direct_write_mode='serial',
+    #printer_port="/dev/tty.usbmodem1411",
+    printer_port="COM6",
+)
+
+
+def read_profilometer(samples=1):
+    samples = np.floor(np.log2(samples))
```

Comment:
I see now, it is to take the number of samples you desire and force it to be a power of 2. So if you put in 18 for samples, it takes the `log2(samples)` and rounds down from 4.169925 to 4, and then 2^4 is 16. Awesome.